distribution/launcher/src/main/java/org/eclipse/sensinact/gateway/launcher/FeatureLauncher.java

```diff
@@ -296,7 +319,13 @@ private void addOrUpdate(String feature, List<String> installProgress) throws Co
         // Start
         for (Bundle b : installed) {
             try {
-                b.start();
+                BundleRevision rev = b.adapt(BundleRevision.class);
+                if (rev == null || (rev.getTypes() & BundleRevision.TYPE_FRAGMENT) == 0) {
```

Comment:
```suggestion
                if (rev != null && (rev.getTypes() & BundleRevision.TYPE_FRAGMENT) == 0) {
```
The revision will only be null for UNINSTALLED bundles which we should also not try to start!